service/src/main/kotlin/fi/espoo/evaka/backupcare/BackupCareQueries.kt

```diff
@@ -113,3 +113,105 @@ WHERE id = :id
 """
 ).bind("id", id)
     .execute()
+
+fun Database.Read.getBackupCareChildId(id: BackupCareId): ChildId = createQuery(
+    // language=SQL
+    """
+    SELECT child_id FROM backup_care WHERE id = :id
+    """.trimIndent()
+)
+    .bind("id", id)
+    .mapTo<ChildId>()
+    .one()
+
+fun Database.Transaction.deleteOrMoveConflictingBackupCares(childId: ChildId, oldRange: FiniteDateRange, newRange: FiniteDateRange?) {
+    if (newRange == null) {
+        createUpdate(
+            // language=SQL
+            """
+            DELETE FROM backup_care
+            WHERE child_id = :childId AND :range @> daterange(start_date, end_date, '[]')
+            """.trimIndent()
+        )
+            .bind("childId", childId)
+            .bind("range", oldRange)
+            .execute()
+
+        createUpdate(
+            // language=SQL
+            """
+            UPDATE backup_care
+            SET end_date = lower(:range) - INTERVAL '1 day'
+            WHERE child_id = :childId AND lower(:range) BETWEEN start_date AND end_date
+            """.trimIndent()
+        )
+            .bind("childId", childId)
+            .bind("range", oldRange)
+            .execute()
+
+        createUpdate(
+            // language=SQL
+            """
+            UPDATE backup_care
+            SET start_date = upper(:range) + INTERVAL '1 day'
```

Comment:
... and the gotcha is that for dateranges, `upper` always returns the *exclusive* endpoint, even if the range was created as inclusive `daterange(..., ..., '[]')`. This happens because inclusivity/exclusivity info is not stored in the range itself, so it's only used during creation and all date ranges are `'[)` internally.

In other words, there seems to be an off-by-one bug here.

Reply:
Thanks, I realised this issue in the else-branch below (when the placement is just moved, not deleted), but didn't update these queries too.